File: src/tools/featureFlags/getAll.ts
import type { z } from "zod";
import type { Context, Tool } from "../types";
import { FeatureFlagGetAllSchema } from "../../schema/tool-inputs";

const schema = FeatureFlagGetAllSchema;

type Params = z.infer<typeof schema>;

export const getAllHandler = async (context: Context, _params: Params) => {
	const projectId = await context.getProjectId();

	const flagsResult = await context.api.featureFlags({ projectId }).list();
	if (!flagsResult.success) {
		throw new Error(`Failed to get feature flags: ${flagsResult.error.message}`);
	}

	return { content: [{ type: "text", text: JSON.stringify(flagsResult.data) }] };
};

const tool = (): Tool<typeof schema> => ({
	name: "feature-flag-get-all",
	description: `
        - Use this tool to get all feature flags in the project.
    `,
	schema,
	handler: getAllHandler,
});

export default tool;

File: python/tools/feature_flags/get_all.py
import json

from api.client import is_error, is_success
from schema.tool_inputs import FeatureFlagGetAllSchema
from tools.tool_definitions import get_tool_definition
from tools.types import Context, Tool, ToolResult


async def get_all_feature_flags_handler(context: Context, _params: FeatureFlagGetAllSchema) -> ToolResult:
    project_id = await context.get_project_id()

    flags_result = await context.api.feature_flags(project_id).list()

    if is_error(flags_result):
        raise Exception(f"Failed to get feature flags: {flags_result.error}")

    assert is_success(flags_result)

    flags_data = [flag.model_dump() for flag in flags_result.data]

    return ToolResult(content=json.dumps(flags_data))


def get_all_feature_flags_tool() -> Tool[FeatureFlagGetAllSchema]:
    definition = get_tool_definition("feature-flag-get-all")

    return Tool(
        name="feature-flag-get-all",
        description=definition["description"],
        schema=FeatureFlagGetAllSchema,
        handler=get_all_feature_flags_handler,
    )
